indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskClient.java

```diff
@@ -376,10 +385,47 @@ public ListenableFuture<DateTime> getStartTimeAsync(final String id)
     return doAsync(() -> getStartTime(id));
   }
 
-
   public ListenableFuture<Map<PartitionIdType, SequenceOffsetType>> pauseAsync(final String id)
   {
-    return doAsync(() -> pause(id));
+    ListenableFuture<Map<PartitionIdType, SequenceOffsetType>> future = doAsync(() -> pause(id));
+    pausingTaskFutureMap.put(id, new TaskPauseControlInfo(future, true));
+    return future;
+  }
+
+  public void cancelTaskPauseRequests()
+  {
+    for (Map.Entry<String, TaskPauseControlInfo> entry : pausingTaskFutureMap.entrySet()) {
+      String taskId = entry.getKey();
+      TaskPauseControlInfo taskPauseControl = entry.getValue();
+
+      if (!taskPauseControl.getFuture().isDone()) {
+        this.stopPausingTask(taskId, taskPauseControl);
+        log.info("Cancel unfinished pause task [%s]", taskId);
+      } else {
+        log.info("Finished pause task [%s]", taskId);
+      }
+    }
+
+    pausingTaskFutureMap.clear();
+  }
+
+  @VisibleForTesting
+  public int getPauseFutureSize()
```

Comment:
We shouldn't need to expose this. There should be other ways to verify that the pause has been cancelled for a given taskId. 

For example, I can see is that while retrying to pause, we keep polling on the status (`getStatus`). In the test, we can assert that if pause requests have been cancelled, `getStatus` is not called anymore.

Reply:
removed.